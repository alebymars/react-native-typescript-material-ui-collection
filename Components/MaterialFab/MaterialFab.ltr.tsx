import * as React from "react";
import {MKButton,mdl} from 'react-native-material-kit';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'
import metrics from "../../../Themes/Metrics";
import {MaterialFabProps} from "./index";

export default (props : MaterialFabProps) => {
  return (
    <MKButton
      width={props.size}
      height={props.size}
      borderRadius={props.size / 2}
      backgroundColor={props.color}
      style={{alignItems:'center',justifyContent:'center'}}
      onPress={props.onPress}
    >
      {props.isLoading ?
        <mdl.Spinner
          strokeColor={props.iconColor}
        />
        :
        <MaterialCommunityIcons
          name={props.icon} size={metrics.regularMaterialFab.iconSize} color= {props.iconColor}
        />
        }

    </MKButton>
  );
}
