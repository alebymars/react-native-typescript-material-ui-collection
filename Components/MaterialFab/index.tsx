import * as React from "react";
import Ripple from 'react-native-material-ripple';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'
import {MaterialProgress} from "../..";

export interface MaterialFabProps {
  icon?: React.ReactNode,
  color?:string,
  iconColor?: string,
  onPress?():void,
  iconName:string,
  mini?:boolean,
  position:string,//{'bottomLeft','topRight','topLeft','bottomRight'}
  progress?:boolean
}

export default (props: MaterialFabProps) =>
  <Ripple
    rippleColor={props.iconColor}
    onPress={()=>props.onPress()}
    style={{borderRadius: props.mini?20:28,
      height:props.mini? 40 : 56,
      width: props.mini? 40 : 56,
      backgroundColor:props.color,
      overflow:'hidden',
      zIndex:900,
      elevation:16,
      shadowColor: 'black',
      shadowOpacity: 0.4,
      shadowRadius: 0.4,
      shadowOffset: {
            width: 0.4,
            height: 0.4
      },
      left:props.position === "bottomLeft"|| props.position === "topLeft" ? 16:undefined,
      right:props.position === "bottomRight"|| props.position === "topRight" ? 16:undefined,
      top:props.position === "topRight"|| props.position === "topLeft" ? 16:undefined,
      bottom:props.position === "bottomRight"|| props.position === "bottomLeft" ? 16:undefined,
      position:'absolute',
      alignItems: 'center',
      justifyContent: 'center'}}>
    {props.icon ? props.icon :
      (!props.progress && props.iconName) &&
      <MaterialCommunityIcons
        style={{margin:8}} name={props.iconName} size={24} color= {props.iconColor}
      />
    }
    {props.progress &&
    <MaterialProgress color={props.iconColor} small/>
    }
  </Ripple>;
