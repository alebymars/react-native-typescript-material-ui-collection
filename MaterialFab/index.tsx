import Ltr from "./MaterialFab.ltr";
import Rtl from "./MaterialFab.rtl";
import {connect} from "react-redux";
import * as React from "react";

export interface MaterialFabProps {
  size: number,
  color: string,
  onPress?(): void,
  iconColor: string,
  icon: string,
  isLoading?: boolean,
  isRtl?:boolean
}

export const Direction = (props:MaterialFabProps) => props.isRtl ? <Rtl {...props} /> : <Ltr {...props}/>;

const mapStateToProps = state => {
  return {
    isRtl: state.appSettings.isRtl
  };
};

export default connect(mapStateToProps)(Direction);
