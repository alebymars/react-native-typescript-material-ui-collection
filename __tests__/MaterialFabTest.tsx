import "react-native";
import * as React from "react";
import { shallow } from "enzyme";
import * as renderer from "react-test-renderer";
import {Direction as MaterialFab} from "../MaterialFab";

test("render without crash", () => {
  const tree = renderer.create(
    <MaterialFab onPress={() => {}} color={'blue'} iconColor='white' size={52} icon={'check'} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});


test("onPress", () => {
  let i = 0; // i guess i could have used sinon here too... less is more i guess
  const onPress = () => i++;
  const wrapperPress = shallow(
    <MaterialFab onPress={onPress} color={'blue'} iconColor='white' size={52} icon={'check'} />
  );

  expect(wrapperPress.prop("onPress")).toBe(onPress); // uses the right handler
  expect(i).toBe(0);
  wrapperPress.simulate("press");
  expect(i).toBe(1);
});
