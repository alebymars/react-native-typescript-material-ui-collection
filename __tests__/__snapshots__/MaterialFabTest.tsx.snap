// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crash 1`] = `
<MKTouchable
  accessible={true}
  backgroundColor="blue"
  borderRadius={26}
  enabled={true}
  height={52}
  maskBorderRadius={2}
  maskColor="rgba(255,255,255,0.15)"
  maskDuration={200}
  maskEnabled={true}
  onChange={[Function]}
  onLayout={[Function]}
  onPress={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTouch={[Function]}
  pointerEvents="box-only"
  rippleColor="rgba(255,255,255,0.2)"
  rippleDuration={200}
  rippleLocation="tapLocation"
  shadowAniEnabled={true}
  style={
    Array [
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {},
      ],
      Object {
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
      },
    ]
  }
  width={52}
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    ?
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255,255,255,0.15)",
        "borderRadius": 0,
        "height": 1,
        "left": -0,
        "opacity": 0,
        "overflow": "hidden",
        "position": "absolute",
        "top": -0,
        "width": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.2)",
          "borderRadius": 0,
          "height": 0,
          "left": 0,
          "top": 0,
          "transform": Array [
            Object {
              "scale": 0,
            },
          ],
          "width": 0,
        }
      }
    />
  </View>
</MKTouchable>
`;
